test/fluree/db/query/construct_test.clj

```diff
@@ -0,0 +1,107 @@
+              "@graph"
+              [{"@id" "ex:bbob", "label" ["Billy Bob"]}
+               {"@id" "ex:fbueller", "label" ["Ferris Bueller"]}
+               {"@id" "ex:jbob", "label" ["Jenny Bob"]}
+               {"@id" "ex:jdoe", "label" ["Jane Doe"]}]}
+             @(fluree/query db1 {"@context" context
+                                 "where" [{"@id" "?s" "person:fullName" "?fullName"}]
+                                 "construct" [{"@id" "?s" "label" "?fullName"}]}))))
+    (testing "nil context"
+      (is (= {"@graph"
+              [{"@id" "http://example.org/bbob", "ex:label" ["Billy Bob"]}
+               {"@id" "http://example.org/fbueller", "ex:label" ["Ferris Bueller"]}
+               {"@id" "http://example.org/jbob", "ex:label" ["Jenny Bob"]}
+               {"@id" "http://example.org/jdoe", "ex:label" ["Jane Doe"]}]}
+             @(fluree/query db1 {"@context" nil
+                                 "where" [{"@id" "?s" "http://example.org/Person#fullName" "?fullName"}]
+                                 "construct" [{"@id" "?s" "ex:label" "?fullName"}]}))))
+    (testing "multiple clauses"
+      (is (= {"@context" {"person" "http://example.org/Person#", "ex" "http://example.org/" "id" "@id"}
+              "@graph"
+              [{"id" "ex:bbob", "name" ["Billy Bob"], "num" [23]}
+               {"id" "ex:jbob", "name" ["Jenny Bob"], "num" [0 3 5 6 7 8 9]}
+               {"id" "ex:jdoe", "name" ["Jane Doe"], "num" [3 7 42 99]}]}
+             @(fluree/query db1 {"@context" (assoc context "id" "@id")
+                                 "where" [{"@id" "?s" "person:fullName" "?fullName"}
+                                          {"@id" "?s" "person:favNums" "?num"}]
+                                 "construct" [{"@id" "?s" "name" "?fullName"}
+                                              {"@id" "?s" "num" "?num"}]}))))
+    (testing "multiple clauses, different subjects"
+      (is (= {"@context" {"person" "http://example.org/Person#", "ex" "http://example.org/"}
+              "@graph"
+              [{"@id" "ex:bbob", "myname" ["Billy Bob"], "friendname" ["Jenny Bob"]}
+               {"@id" "ex:jbob", "name" ["Jenny Bob"], "num" [0 3 5 6 7 8 9]}]}
+             @(fluree/query db1 {"@context" context
+                                 "where" [{"@id" "?s" "person:fullName" "?fullName"}
+                                          {"@id" "?s" "person:friend" "?friend"}
+                                          {"@id" "?friend" "person:fullName" "?friendName"}
+                                          {"@id" "?friend" "person:favNums" "?friendNum"}]
+                                 "construct" [{"@id" "?s" "myname" "?fullName"}
+                                              {"@id" "?s" "friendname" "?friendName"}
+                                              {"@id" "?friend" "name" "?friendName"}
+                                              {"@id" "?friend" "num" "?friendNum"}]}))))
+    #_(testing "bnode template"
```

Comment:
Using bnodes in construct templates produces nondeterministic responses since we process results in parallel. I don't have a good solution for this yet, so the flaky tests are commented out.